src/OperationResponse.php

```diff
@@ -0,0 +1,230 @@
+ * Operations API, which is used by the OperationResponse object. If
+ * more control is required, it is possible to make calls against the
+ * Operations API directly instead of via the OperationResponse object
+ * using an OperationsClient instance.
+ */
+class OperationResponse
+{
+    const DEFAULT_POLLING_INTERVAL = 1.0;
+
+    private $operationName;
+    private $operationsClient;
+    private $operationReturnType;
+    private $metadataReturnType;
+    private $lastProtoResponse;
+    private $deleted = false;
+
+    /**
+     * OperationResponse constructor.
+     *
+     * @param string $operationName
+     * @param \Google\Longrunning\OperationsClient $operationsClient
+     * @param array $options {
+     *                       Optional. Options for configuring the Operation response object.
+     *
+     *     @type string $operationReturnType The return type of the longrunning operation.
+     *     @type string $metadataReturnType The type of the metadata returned in the Operation response.
+     *     @type \google\longrunning\Operation $lastProtoResponse A response already received from the server.
+     * }
+     */
+    public function __construct($operationName, $operationsClient, $options = [])
+    {
+        $this->operationName = $operationName;
+        $this->operationsClient = $operationsClient;
+        if (isset($options['operationReturnType'])) {
+            $this->operationReturnType = $options['operationReturnType'];
+        }
+        if (isset($options['metadataReturnType'])) {
+            $this->metadataReturnType = $options['metadataReturnType'];
+        }
+        if (isset($options['lastProtoResponse'])) {
+            $this->lastProtoResponse = $options['lastProtoResponse'];
+        }
+    }
+
+    /**
+     * Check whether the operation has completed.
+     *
+     * @return bool
+     */
+    public function isDone()
+    {
+         return (is_null($this->lastProtoResponse) || is_null($this->lastProtoResponse->getDone()))
+             ? false
+             : $this->lastProtoResponse->getDone();
+    }
+
+    /**
+     * Get the formatted name of the operation
+     *
+     * @return string The formatted name of the operation
+     */
+    public function getName()
+    {
+        return $this->operationName;
+    }
+
+    /**
+     * Poll the server in a loop until the operation is complete.
+     *
+     * Return true if the operation completed successfully, otherwise return
+     * false.
+     *
+     * The $pollSettings optional argument can be used to control the polling loop.
+     *
+     * @param array $options {
+     *                       Options for configuring the polling behaviour.
+     *
+     *     @type float $pollingIntervalSeconds The polling interval to use, in seconds.
+     *                                          Default: 1.0
+     * }
+     * @return boolean Indicates if the operation completed successfully.
```

Comment:
Personally, I tend to prefer the short version of variable names in docblocks. Since the short version was used in `isDone()` above WDYT about the same here?